src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -167,6 +164,106 @@ public ResponseEntity<byte[]> getTemplateFile(@PathVariable ProgrammingLanguage
+        try {
+            String fileName = buildAeolusTemplateName(projectTypePrefix, staticAnalysis, sequentialRuns, testCoverage);
+            Resource fileResource = resourceLoaderService.getResource(Path.of("templates", "aeolus", languagePrefix, fileName));
+            if (!fileResource.exists()) {
+                throw new IOException("File " + fileName + " not found");
+            }
+            byte[] fileContent = IOUtils.toByteArray(fileResource.getInputStream());
+            String yaml = new String(fileContent, StandardCharsets.UTF_8);
+            String json = convertYamlToJson(yaml);
+            HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.setContentType(MediaType.APPLICATION_JSON);
+            return new ResponseEntity<>(json, responseHeaders, HttpStatus.OK);
+        }
+        catch (IOException ex) {
+            log.debug("Error when retrieving aeolus template file : {}", ex.getMessage());
+            HttpHeaders responseHeaders = new HttpHeaders();
+            return new ResponseEntity<>(null, responseHeaders, HttpStatus.NOT_FOUND);
+        }
+    }
+
+    /**
+     * Returns the file content of the template file for the given language and project type with the different options
+     *
+     * @param projectTypePrefix The project type for which the template file should be returned. If omitted, a default depending on the language will be used.
+     * @param staticAnalysis    whether the static analysis template should be used
+     * @param sequentialRuns    whether the sequential runs template should be used
+     * @param testCoverage      whether the test coverage template should be used
+     * @return The requested file, or 404 if the file doesn't exist
+     */
+    private String buildAeolusTemplateName(String projectTypePrefix, Boolean staticAnalysis, Boolean sequentialRuns, Boolean testCoverage) {
+        List<String> fileNameComponents = new ArrayList<>();
+        if (staticAnalysis) {
+            fileNameComponents.add("static");
+        }
+        if (sequentialRuns) {
+            fileNameComponents.add("sequential");
+        }
+        if (testCoverage) {
+            fileNameComponents.add("coverage");
+        }
+        String accumulator = "default";
+        if (!projectTypePrefix.isEmpty()) {
+            accumulator = projectTypePrefix;
+        }
+        return fileNameComponents.stream().reduce(accumulator, (a, b) -> a + "_" + b) + ".yaml";
+    }
+
+    /**
+     * Converts a YAML string to a JSON string for easier communication with the client
+     *
+     * @param yaml YAML string
+     * @return JSON string
+     */
+    private String convertYamlToJson(String yaml) throws JsonProcessingException {
+        ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+        Object obj = yamlReader.readValue(yaml, Object.class);
+
+        ObjectMapper jsonWriter = new ObjectMapper();
+        return jsonWriter.writeValueAsString(obj);
+    }
```

Comment:
This should probably be part of an `AelousTemplateService` (exact name to be determined) since the Resource/RESTController should not contain much logic but only check permissions and convert the client data into a usable format for the service (separation of concerns).

Reply:
Why would you add it to a new service? These methods serve static files, the only logic is building the filename and converting yaml to json for a more client friendly handling.

Reply:
Second look: you are right, it is fine as-is. :+1: